packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { RemoteParticipant } from '@azure/communication-calling';
+import { _formatString } from '@internal/acs-ui-common';
+import { useEffect, useState } from 'react';
+import { useLocale } from '../../localization';
+import { useAdapter } from '../adapter/CallAdapterProvider';
+
+/**
+ * Delay value for when a participant has left or joined
+ *
+ * this is to allow the narrator some room so it isnt stuttering whenever there is a change
+ * in participants.
+ */
+const PARTICIPANT_ANNOUNCEMENT_DELAY = 500;
+
+/**
+ * sets the announcement string whenever a Participant comes or goes from a call to be
+ * used by the system narrator.
+ *
+ * @returns string to be used by the narrator and Announcer component
+ *
+ * @internal
+ */
+export const useParticipantChangedAnnouncement = (): string => {
+  const adapter = useAdapter();
+  const locale = useLocale().strings.call;
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  /**
+   * state to track whether there is currently a timer set in the useParticipantChangedAnnouncement hook
+   */
+  const [timeoutState, setTimeoutState] = useState<ReturnType<typeof setTimeout>>();
+
+  useEffect(() => {
+    const setParticipantEventString = (string: string): void => {
+      setAnnouncerString('');
```

Comment:
this still has the issues where if a participant joins or leaves before the `PARTICIPANT_ANNOUNCEMENT_DELAY` elapses, the previous announcement won't be read out